homeassistant/components/sleepiq/number.py

```diff
@@ -148,6 +191,8 @@ def __init__(
 
         self._attr_name = description.get_name_fn(bed, device)
         self._attr_unique_id = description.get_unique_id_fn(bed, device)
+        if description.icon:
+            self._attr_icon = description.icon
```

Comment:
This shouldnt be needed?

Reply:
It could be just  `self._attr_icon = description.icon` but the linter complains because `icon` could be `None`

Reply:
Yea but the standard implementation of the `icon` property first check if entity description is set, if so, it will get the icon from there. So I'm not sure why this wouldn't work here. (This would require removing both lines)

Reply:
In `SleepIQBedEntity` the line `_attr_icon = ICON_OCCUPIED` is overriding it.  Without it being set here it doesn't get set